### PR TITLE
chore(proto): format proto with buf

### PR DESCRIPTION
**Pull request by janisz**
## Description

This PR introduces `buf format` to keep our protos formatted in the same way. 

> [The time saved by not arguing over spaces and newlines is worth all the time spent defining a standard format and writing this rather difficult piece of code to automate it.](https://youtu.be/yE5Tpp2BSGw?si=3H9G3YKsmASL93mi&t=634) 

This is required as next step will be to drop gogo proto that will be easier to handle if all protos looks similar.

See: 
- https://github.com/stackrox/stackrox/pull/9380
- https://buf.build/blog/introducing-buf-format

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

CI

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by connorgorman — COMMENTED** [review]
This may just be a personal preference, but I'm not a huge fan of how buf formats the individual entries and felt like aligning the numbers altogether makes it more readable

**Comment by connorgorman** [comment]
Can you elaborate on how this makes removing gogo easier?

**Comment by dashrews78** [comment]
> This may just be a personal preference, but I'm not a huge fan of how buf formats the individual entries and felt like aligning the numbers all together makes it more readable

+1

**Comment by janisz** [comment]
> Can you elaborate on how this makes removing gogo easier?

I forked buf and hacked it to convert gogotags into @gotag https://github.com/janisz/buf/commit/61be349ba30339553eb179788c1529dd995e69b2

**Review by mtodor — COMMENTED** [review]
It looks good. Found one unresolved merge (I guess).

**Comment by mtodor** [comment]
I did a review of this PR. Information about the process is shared in Slack.

File: central/complianceoperator/v2/rules/store/postgres/store.go
Comment (mtodor) [inline central/complianceoperator/v2/rules/store/postgres/store.go:1]: If I'm not wrong, this file (and test also) should not be a part of this PR.

File: proto/api/v1/search_service.proto [hunk 1]
```diff
@@ -1,90 +1,89 @@
...
-    COMPLIANCE_CHECK_RESULTS       = 57;
-    BLOB                           = 58;
-    // Was PRODUCT_USAGE previously, now replaced with ADMINISTRATION_USAGE
-    reserved                          59;
-    ADMINISTRATION_EVENTS           = 60;
-    COMPLIANCE_SCAN_CONFIG_STATUS   = 61;
-    ADMINISTRATION_USAGE            = 62;
-    COMPLIANCE_PROFILES             = 63;
-    COMPLIANCE_PROFILE_CLUSTER_EDGE = 64;
-    COMPLIANCE_RULES                = 65;
```
Comment (mtodor) [inline proto/api/v1/search_service.proto:78]: We are missing the recently added option. It was probably lost during the rebase.

File: proto/internalapi/central/compliance_operator.proto [hunk 2]
```diff
@@ -1,226 +1,221 @@
...
-    string                         id           = 2;
-    string                         name         = 3;
-    string                         rule_type    = 4;
-    ComplianceOperatorRuleSeverity severity     = 5;
-    map<string, string>            labels       = 6;
-    map<string, string>            annotations  = 7;
-    string                         title        = 8;
-    string                         description  = 9;
-    string                         rationale    = 10;
-    message Fix {
```
Comment (mtodor) [inline proto/internalapi/central/compliance_operator.proto:199]: `message Fix` - is missing in formatted proto.

File: proto/storage/compliance_operator_v2.proto [hunk 3]
```diff
@@ -1,197 +1,186 @@
...
-    string                description      = 9;
-    string                rationale        = 10;
-    message Fix {
-        string platform   = 1;
-        string disruption = 2;
-    }
-    repeated Fix          fixes            = 11;
-    string                warning          = 12;
-    repeated RuleControls controls         = 13;
-    string                cluster_id       = 14 [(gogoproto.moretags) = 'search:"Cluster ID,hidden" sql:"fk(Cluster:id),no-fk-constraint,type(uuid)"'];
```
Comment (mtodor) [inline proto/storage/compliance_operator_v2.proto:82]: A new field is missing in the formatted proto.

File: proto/storage/image_integration.proto [hunk 4]
```diff
@@ -1,150 +1,149 @@
...
-        ClairifyConfig    clairify   = 8;
-        ScannerV4Config   scanner_v4 = 23;
-        DockerConfig      docker     = 9;
-        QuayConfig        quay       = 10;
-        ECRConfig         ecr        = 11;
-        GoogleConfig      google     = 13;
-        ClairConfig       clair      = 14;
-        ClairV4Config     clair_v4   = 22;
-        IBMRegistryConfig ibm        = 20;
-        AzureConfig       azure      = 24;
```
Comment (mtodor) [inline proto/storage/image_integration.proto:32]: `AzureConfig` is missing in the formatted proto.

**Review by mtodor — COMMENTED** [review]
I think that the last rebase didn't go well.

**Review by mtodor — DISMISSED** [review]
Checked it! It looks good!

**Review by dashrews78 — APPROVED** [review]
I'm good with it.  Due to size, I recommend multiple approvals though.

**Review by 0x656b694d — APPROVED** [review]

**Review by mtodor — APPROVED** [review]
Looks good. It would be good to get at least one successful `qa-e2e-test` (on any cluster).

**Pull request merged**